### DISCUSSION TITLE
Is there a way to override the configmap namespace when injecting the java agent?

### DISCUSSION THREAD

**Discussion by ghost** [apache/skywalking#8597] [Q&A] [answered]
I have skywalking-swck operator deployed in a custom namespace `skywalking`. All items seem to be correctly deployed and I don't see any apparent error.

When trying to configure a deployment/namespace to inject the agent, I configure the specified labels in both namespace and deployment, but the java agent doesn't get deployed, since its trying to find the configmap in the default namespace `skywalking-swck-system`.

As workarounds I could think of either changing the value of `DefaultConfigmapNamespace` in `configmap.go` and build the application or manually create the configmap in the desired namespace, but I was wondering if is there a way to override its value from a label/annotation/etc or I'm forced to have it installed in a namespace called `skywalking-swck-system`?

**Comment by wu-sheng** [comment]
@dashanji Your suggestion?

**Comment by dashanji** [comment, marked as answer]
@Dwosky 
Hi, thanks for your interest in skywalking-swck. 

Firstly, you have deployed the operator in a custom namespace `skywalking `, but the default configmap is still created in `skywalking-swck-system`, I think it's a bug because the namespace in the `config/configmap/configmap.yaml` should be `system`, so that the operator and the default configmap could be created in the same namespace. Thanks again for figuring it out.

Secondly, in our design, the default configmap should be in the default system namespace as the operator. If we need to override it when we changed the default system namespace, I think we can label the system namespace and update the `DefaultConfigmapNamespace ` by getting the system namespace of the specified label in the `configmap_controller.go`. By the way, do you have any interest in the feature?

**Reply by ghost** [reply]
Hello @dashanji, thanks for the quick reply.

I think there has been a misunderstanding regarding the first point, sorry if I haven't explained it correctly. In my case all objects are inside the `skywalking` custom namespace I've set, including the configmap for the java agent. The problem itself is that the core of the app tries to look for the configmap in `skywalking-swck-system` namespace and there isn't an elegant way of overriding that behavior.

I have been messing with skywalking-swck for just a couple of days for a PoC, so I'm not sure if its really a feature someone aside that me would like. The thing is that seeing the namespace definition in the `default/kustomization.yaml` configuration file I though the application didn't had any hard ties to the namespace itself and you could change it.

**Reply by dashanji** [reply]
Yep, the core problem is that the default system namespace is hard-coded and the behavior to find the default configmap is the current design. when we deploy the operator in a custom namespace like `skywalking`, as I mentioned before, a possible way is to find the namespace where the operator is deployed and update it. For example, if we need to deploy the operator in a namespace called `skywalking`, we need to label the namespace  `skywalking` like `skywalking-system: skywalking`, and the operator should find the configmap in the default namespace `skywalking-swck-system` firstly, if can't find,  the operator needs to find the namespace with the specified label next and update the system namespace.

I think it's an enhancement, as users may need to create the operator in the custom namespace. What I said is only an alternative way, if you have any idea, please feel free to bring it up. In addition, if you meet any problems in using skywalking-swck, please feel free to ask, we will try to solve it and make the operator better.

**Reply by ghost** [reply]
For the time being I'll keep on the default namespace to make the setup a little simpler to maintain. Thanks for the response.
